packages/ember-debug/lib/deprecate.js

```diff
@@ -0,0 +1,93 @@
+/*global __fail__*/
+
+import Ember from 'ember-metal/core';
+import EmberError from 'ember-metal/error';
+import Logger from 'ember-metal/logger';
+import { registerHandler as genericRegisterHandler, invoke } from 'ember-debug/handlers';
+
+export function registerHandler(handler) {
+  genericRegisterHandler('deprecate', handler);
+}
+
+function formatMessage(_message, options) {
+  let message = _message;
+
+  if (options && options.id) {
+    message = message + ` [deprecation id: ${options.id}]`;
+  }
+
+  if (options && options.url) {
+    message += ' See ' + options.url + ' for more details.';
+  }
+
+  return message;
+}
+
+registerHandler(function logDeprecationToConsole(message, options) {
+  let updatedMessage = formatMessage(message, options);
+
+  Logger.warn('DEPRECATION: ' + updatedMessage);
+});
+
+registerHandler(function logDeprecationStackTrace(message, options, next) {
+  if (Ember.LOG_STACKTRACE_ON_DEPRECATION) {
+    let stackStr = '';
+    let error, stack;
+
+    // When using new Error, we can't do the arguments check for Chrome. Alternatives are welcome
+    try { __fail__.fail(); } catch (e) { error = e; }
+
+    if (error.stack) {
+      if (error['arguments']) {
+        // Chrome
+        stack = error.stack.replace(/^\s+at\s+/gm, '').
+          replace(/^([^\(]+?)([\n$])/gm, '{anonymous}($1)$2').
+          replace(/^Object.<anonymous>\s*\(([^\)]+)\)/gm, '{anonymous}($1)').split('\n');
+        stack.shift();
+      } else {
+        // Firefox
+        stack = error.stack.replace(/(?:\n@:0)?\s+$/m, '').
+          replace(/^\(/gm, '{anonymous}(').split('\n');
+      }
+
+      stackStr = '\n    ' + stack.slice(2).join('\n    ');
+    }
+
+    let updatedMessage = formatMessage(message, options);
+
+    Logger.warn('DEPRECATION: ' + updatedMessage + stackStr);
+  } else {
+    next(...arguments);
+  }
+});
+
+registerHandler(function raiseOnDeprecation(message, options, next) {
+  let updatedMessage = formatMessage(message);
```

Comment:
I believe this would format the message twice. Once here and (if raise on deprecation is false) once in `logDeprecationToConsole`. Seems like this let should be inside the `if`.